blockchain/chain.go

```diff
@@ -1033,96 +1131,12 @@ func (b *BlockChain) reorganizeChain(detachNodes, attachNodes *list.List) error
 					b.index.SetStatusFlags(dn, statusInvalidAncestor)
 				}
 			}
-			return err
+			return nil, nil, nil, err
 		}
 		b.index.SetStatusFlags(n, statusValid)
-
-		newBest = n
-	}
-
-	// Flush the utxo cache for the block disconnect below.  The disconnect
-	// code assumes that it's directly modifying the database so the cache
-	// will be left in an inconsistent state.  It needs to be flushed beforehand
-	// in order for that to not happen.
-	err := b.db.Update(func(dbTx database.Tx) error {
```

Comment:
Outstanding question is whether we need to do this still?

Reply:
@kcalvinalvin

Reply:
It wasn't needed [here](https://github.com/btcsuite/btcd/commit/a254998bc5e348a227cdb454ffca0e9e3857c4a1) (the PR for when it was added) since the flush on `disconnectBlock` would always ensure that the right state would be saved to disk.

No need for it to be here and it's ok to remove it.